### ISSUE TITLE
IMethodInterceptor's intercept method executes twice.

### ISSUE THREAD

**Issue by kodayashi** [cbeust/testng#1192]
### TestNG Version
6.9.5

### Expected behavior
The intercept method should only be invoked once.

### Actual behavior
The intercept method is invoked twice.

### Is the issue reproductible on runner?
I'm using maven 3.3.9 with eclipse 4.9.2

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [x] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
> Please, share the test case (as small as possible) which shows the issue

```java
// MyListener class
public class MyListener extends TestListener {

	public MyListener() {
	}

	@Override
	public void onTestStart(ITestResult result) {
		super.onTestStart(result);
	}

	@Override
	public void onTestSuccess(ITestResult result) {
		super.onTestSuccess(result);
	}

	@Override
	public void onTestFailure(ITestResult result) {
		super.onTestFailure(result);
	}

	@Override
	public void onTestSkipped(ITestResult result) {
		super.onTestSkipped(result);
	}

	@Override
	public void onTestFailedButWithinSuccessPercentage(ITestResult result) {
		super.onTestFailedButWithinSuccessPercentage(result);
	}

	@Override
	public void onStart(ITestContext context) {
		super.onStart(context);
	}

	@Override
	public void onFinish(ITestContext context) {
		super.onFinish(context);
	}
}
```

```java
// MyInterceptor
public class MyInterceptor implements IMethodInterceptor {
	public MyInterceptor() {
	}

	@Override
	public List<IMethodInstance> intercept(List<IMethodInstance> methods, ITestContext context) {
                System.out.println("In intercept method.");

		List<IMethodInstance> result = new ArrayList<IMethodInstance>();

                 // Add some methods

		return result;
	}

}
```

```java
// BaseTest class
@Listeners(MyListener.class)
public class BaseTest {

	@BeforeSuite(alwaysRun = true)
	public static void BeforeSuite() {
               System.out.println("Before suite.");
	}

	@AfterSuite(alwaysRun = true)
	public static void afterSuite() {
		System.out.println("After suite.");
	}
}
```

```java
// TestVideo class
@Test(groups = { TestConstants.TestNGGroups.MY_GROUP }, description = "My video test")
public class TestVideo extends BaseTest {

	public void testVideo() {

		logger.info("Testing video.");
		Assert.assertTrue(true);
	}
}
```

```xml
// TestNG xml file
<suite name="My Suite" verbose="1" parallel="classes"
	thread-count="1">
	<listeners>
		<listener class-name="com.mycompany.utils.MyInterceptor" />
	</listeners>
	<test name="TestVideo" preserve-order="true">
	    <classes>
	      <class name="com.mycompany.TestVideo"  />
	    </classes>
	</test>
</suite>
```

**Comment by juherr** [comment]
Please, try with the latest version.

**Comment by kodayashi** [comment]
Shoot, I thought I had latest based on tags in the github repo (6.9.5). Looks like they're not tagging releases anymore, which is fine. Checking here showed me that I did not - https://mvnrepository.com/artifact/org.testng/testng.

To my delight, the issue was apparently fixed between 6.9.5 and 6.9.13.6.

Closing!
